### ISSUE TITLE
catboost plot not working for colab

### ISSUE THREAD

**Issue by AnandUgale** [catboost/catboost#985]
Problem: catboost plot not working for colab
catboost version: 0.16.5
Operating System: Windows 10 (64 bit)
CPU: ON
# GPU: colab GPU

**Comment by aamster** [comment]
most likely you have to initialize the notebook for plotting or you are executing code which outputs a plot outside of the notebook. Could you provide more details?

**Comment by AnandUgale** [comment]
!pip install catboost
!pip install ipywidgets
!pip install shap
#!pip install sklearn
!jupyter nbextension enable --py widgetsnbextension

model = CatBoostClassifier(
    iterations=50,
    learning_rate=0.5,
    custom_loss=['AUC', 'Accuracy']
)

model.fit(
    train_pool,
    eval_set=validation_pool,
    verbose=False,
    plot=True
);

not able to see any graphs after enabling plot=True

**Comment by aamster** [comment]
Does indeed seem to be an issue. Plotting in local notebook seems to work. I would look into it but not sure where the relevant code path is...

**Comment by annaveronika** [comment]
This is probably the same problem as https://github.com/catboost/catboost/issues/262

**Comment by ThamaluM** [comment]
I'll add my experience on developing charting libraries for notebooks since this will help contributors to solve the issue.

Those interactive graphs are made by using JavaScript charting libraries. 

There is a little different in normal jupyter-notebooks architecture in contrast to colab notebooks. 

The normal notebooks are made up by several div components whereas google colab uses a set of IFrames to build up notebook.

Therefore charting libraries cannot use the library importing it globally (as outputs come from different IFrames).

Plotly4.x is the only interactive charting library I know that is able to render interactive plots in colab environments.

**Comment by VlasovKirill** [comment]
Unfortunately, Colab does not support widgets, which not included in the ipywidgets package. More information: https://github.com/googlecolab/colabtools/issues/498
